infrastructure/server/fam_admin_management_api.tf

```diff
@@ -0,0 +1,138 @@
+# Looking up a few things so they can be changed for this file in one place only
+
+data "aws_secretsmanager_secret" "db_admin_management_api_creds_secret" {
+  name = aws_secretsmanager_secret.famdb_admin_management_apicreds_secret.name
+}
+
+data "aws_rds_cluster" "admin_management_api_database" {
+  cluster_identifier = var.famdb_cluster_name
+  depends_on = [
+    module.aurora_postgresql_v2
+  ]
+}
+
+data "aws_db_proxy" "admin_management_api_lambda_db_proxy" {
+  name = aws_db_proxy.famdb_proxy_api.name
+}
+
+locals {
+  admin_management_api_lambda_name = "fam-admin-management-api-lambda-${var.target_env}"
+}
+
+resource "aws_iam_role_policy" "fam_admin_management_api_lambda_access_policy" {
+  name   = "${local.admin_management_api_lambda_name}-access-policy"
+  role   = aws_iam_role.fam_admin_management_api_lambda_exec.id
+  policy = <<-EOF
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Action": [
+            "logs:CreateLogGroup",
+            "logs:CreateLogStream",
+            "logs:PutLogEvents",
+            "ec2:CreateNetworkInterface",
+            "ec2:DescribeNetworkInterfaces",
+            "ec2:DeleteNetworkInterface",
+            "ec2:AssignPrivateIpAddresses",
+            "ec2:UnassignPrivateIpAddresses"
+        ],
+        "Resource": "*"
+      },
+      {
+        "Effect": "Allow",
+        "Action": [
+          "secretsmanager:DescribeSecret",
+          "secretsmanager:GetSecretValue"
+        ],
+        "Resource": "${data.aws_secretsmanager_secret.db_admin_management_api_creds_secret.arn}"
+      },
+      {
+        "Effect": "Allow",
+        "Action": [
+          "secretsmanager:DescribeSecret",
+          "secretsmanager:GetSecretValue"
+        ],
+        "Resource": "${data.aws_secretsmanager_secret.fam_oidc_client_id_secret.arn}"
```

Comment:
I have one question, this "data.aws_secretsmanager_secret.fam_oidc_client_id_secret" is defined in "fam_api.tf", it's just get the oidc client id secret, it should be same? do I need to define a separate one in this file? 

```
data "aws_secretsmanager_secret" "fam_oidc_client_id_secret" {
  name = aws_secretsmanager_secret.fam_oidc_client_id_secret.name
}
```

Reply:
We can reuse `data` that's defined, we seem to have no common file for the `data`.
I think it is ok for now just reuse it, we can refactor it later.